hybrid/pkg/agent/agent_test.go

```diff
@@ -0,0 +1,430 @@
+
+	interceptor.payload = nil
+	interceptorOne := interceptor.SpawnInterceptor()
+	interceptorOne.SetPluginName("test_pluginOne")
+	interceptorOne.Send(&payloadOne)
+
+	interceptorTwo := interceptor.SpawnInterceptor()
+	interceptorTwo.SetPluginName("test_pluginTwo")
+	interceptorTwo.Send(&payloadTwo)
+	var messageList common.PluginMessageList = common.PluginMessageList{}
+
+	message1 := interceptorOne.GetMessage()
+	require.Equal(t, []byte(payloadOneData), message1.PluginData, "Could not get message from interceptor")
+	require.Equal(t, "test_pluginOne", message1.PluginName, "Could not get plugin name from interceptor")
+	message2 := interceptorTwo.GetMessage()
+	require.Equal(t, []byte(payloadTwoData), message2.PluginData, "Could not get message from interceptor")
+	require.Equal(t, "test_pluginTwo", message2.PluginName, "Could not get plugin name from interceptor")
+	messageList.Messages = append(messageList.Messages, message1)
+	messageList.Messages = append(messageList.Messages, message2)
+	interceptor.SendCombined(messageList)
+
+	jsonMessage, err := json.Marshal(messageList)
+	require.NoError(t, err, "Error marshalling message list: %w", err)
+	require.Equal(t, &jsonMessage, stream.Payload, "Could not send combined message list to stream")
+}
+
+func TestHybridPluginAgentAidAttestation(t *testing.T) {
+
+	emptyPayload := []byte("")
+	stream := StreamMock{Payload: &emptyPayload}
+
+	pluginOne := new(FakePlugin)
+	pluginTwo := new(FakePlugin)
+	pluginList := []common.Types{
+		{PluginName: "k8s_psat", Plugin: pluginOne},
+		{PluginName: "aws_iid", Plugin: pluginTwo},
+	}
+	interceptorFake := new(InterceptorWrapper)
+	hybridPlugin := HybridPluginAgent{pluginList: pluginList, logger: hclog.Default(), interceptor: interceptorFake}
+
+	aidAttestation := hybridPlugin.AidAttestation(stream)
+	require.NoError(t, aidAttestation, "AidAttestation of hybrid plugin fails")
+	interceptorFake.SetReturnError(true)
+
+	aidAttestation = hybridPlugin.AidAttestation(stream)
+	require.Error(t, aidAttestation, "AidAttestation of hybrid plugin fails")
+
+	hybridPlugin.SetLogger(hclog.Default().Named("test_logger2"))
+	require.Equal(t, "test_logger2", hybridPlugin.logger.Name(), "Could not set hybrid plugin logger")
+
+	expectedError := status.Error(codes.InvalidArgument, "Plugin initialization error")
+	hybridPlugin.initStatus = expectedError
+	aidAttestation = hybridPlugin.AidAttestation(stream)
+	require.EqualError(t, aidAttestation, expectedError.Error(), "AidAttestation of hybrid plugin fails")
+
+	pluginOne = new(FakePlugin)
+	pluginTwo = new(FakePlugin)
+	pluginTwo.returnError = true
+	pluginList = []common.Types{
+		{PluginName: "k8s_psat", Plugin: pluginOne},
+		{PluginName: "aws_iid", Plugin: pluginTwo},
+	}
+	interceptorFake = new(InterceptorWrapper)
+	hybridPlugin = HybridPluginAgent{pluginList: pluginList, logger: hclog.Default(), interceptor: interceptorFake}
```

Comment:
I'm not sure I agree with you, because in this test we're aiming to make sure that if one of the plugins fail during the aidattestation, that is called by reflection, the whole logic captures that fail and shuts down the plugin. If we initialized without any modificiations to make it fail, we would have to tweak the real implementation for it to fail.